src/main/java/com/hubspot/jinjava/lib/tag/eager/EagerToken.java

```diff
@@ -15,12 +15,14 @@
   private final Set<String> setDeferredWords;
 
   private final String importResourcePath;
+  private final String currentMacroFunction;
 
   public EagerToken(Token token, Set<String> usedDeferredWords) {
     this.token = token;
     this.usedDeferredWords = usedDeferredWords;
     this.setDeferredWords = Collections.emptySet();
     importResourcePath = acquireImportResourcePath();
+    currentMacroFunction = acquireCurrentMacroFunction();
```

Comment:
We track the current macro function so that we know when to stop deferring values. We only want to defer up to the current macro functions context, and not and of the higher-level parents so that it doesn't become deferred in the code that runs outside the macro function's scope.